src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -0,0 +1,1447 @@
+#include "algorithms/synthesis/syrec_synthesis.hpp"
+
+#include "core/functions/add_gates.hpp"
+#include "core/syrec/expression.hpp"
+#include "core/syrec/program.hpp"
+#include "core/syrec/variable.hpp"
+#include "core/utils/timer.hpp"
+
+#include <boost/dynamic_bitset.hpp>
+#include <boost/format.hpp>
+#include <boost/range/adaptors.hpp>
+#include <cmath>
+#include <functional>
+#include <memory>
+#include <numeric>
+
+namespace syrec {
+    static std::stack<unsigned>               exp_opp;
+    static std::stack<std::vector<unsigned>>  exp_lhss, exp_rhss;
+    bool                                      sub_flag = false;
+    static std::vector<unsigned>              op_vec, assign_op_vector, exp_op_vector;
+    static std::vector<std::vector<unsigned>> exp_lhs_vector, exp_rhs_vector;
+
+    struct annotater {
+        explicit annotater(circuit& circ, const std::stack<applications::statement::ptr>& stmts):
+            _circ(circ),
+            _stmts(stmts) {}
+
+        // Operator needs this signature to work
+        void operator()(gate& g) const {
+            if (!_stmts.empty()) {
+                _circ.annotate(g, "lno", std::to_string(_stmts.top()->line_number()));
+            }
+        }
+
+    private:
+        circuit&                                        _circ;
+        const std::stack<applications::statement::ptr>& _stmts;
+    };
+
+    // Helper Functions for the synthesis methods
+    standard_syrec_synthesizer::standard_syrec_synthesizer(circuit& circ, const applications::program& prog [[maybe_unused]]):
+        _circ(circ) {
+        free_const_lines_map.insert(std::make_pair(false, std::vector<unsigned>()));
+        free_const_lines_map.insert(std::make_pair(true, std::vector<unsigned>()));
+
+        // root anlegen
+        cct_man.current = add_vertex(cct_man.tree);
+        cct_man.root    = cct_man.current;
+        // Blatt anlegen
+        cct_man.current                                             = add_vertex(cct_man.tree);
+        get(boost::vertex_name, cct_man.tree)[cct_man.current].circ = std::make_shared<circuit>();
+        get(boost::vertex_name, cct_man.tree)[cct_man.current].circ->gate_added.connect(annotater(*get(boost::vertex_name, cct_man.tree)[cct_man.current].circ, _stmts));
+        add_edge(cct_man.root, cct_man.current, cct_man.tree);
+    }
+
+    void standard_syrec_synthesizer::set_settings(const properties::ptr& settings) {
+        variable_name_format = get<std::string>(settings, "variable_name_format", "%1$s%3$s.%2$d");
+    }
```

Comment:
If the `variable_name_format` member is eliminated, then this function can be removed as well.